src/artifactContentRenderers/artifactContentRendering.ts

```diff
@@ -0,0 +1,47 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ */
+
+import * as vscode from "vscode";
+import * as sarif from "sarif";
+import { BinaryArtifactContentRenderer } from "./binaryArtifactContentRenderer";
+import { TextArtifactContentRenderer } from "./textArtifactContentRenderer";
+import { RenderingArtifactContentRenderer } from "./renderingArtifactContentRenderer";
+
+export interface ArtifactContentRenderer {
```

Comment:
This design pattern promotes an unnecessary allocation of short-lifespan objects.

Reply:
It's not creating any objects (unless I'm missing something). There is an array of functions (that are static in their implementation). No objects are created unless the renderer supports the artifact content.

In effect, it is simply splitting the "if check" that used to exist in one function into three smaller functions.

---
In reply to: [411735831](https://github.com/microsoft/sarif-vscode-extension/pull/252#discussion_r411735831) [](ancestors = 411735831)